source/write/transactions.txt

```diff
@@ -0,0 +1,170 @@
+   ``MongoDatabase`` or ``MongoCollection``) that created it. Using a
+   ``ClientSession`` with a different ``MongoClient`` results in operation
+   errors.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``sample_restaurants.restaurants`` collection
+from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+free MongoDB Atlas cluster and load the sample datasets, see the
+:ref:`<pymongo-get-started>` tutorial.
+
+Methods
+-------
+
+After you start a session by using the ``start_session()`` method, you can manage 
+the session state by using the methods provided by the returned ``ClientSession``.
+
+.. list-table::
+   :widths: 25 75
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``start_transaction()``
+     - | Starts a new transaction, configured with the given options, on
+         this session. Returns an error if there is already
+         a transaction in progress for the session. To learn more about
+         this method, see the :manual:`startTransaction() page
+         </reference/method/Session.startTransaction/>`  in the Server manual.
+       |
+       | **Parameters**: ``read_concern``, ``write_concern``, ``read_preference``, ``max_commit_time_ms``
+       | **Return Type**: ``ContextManager``
+
+   * - ``abort_transaction()``
+     - | Ends the active transaction for this session. Returns an
+         error if there is no active transaction for the session or the
+         transaction has been committed or ended. To learn more about
+         this method, see the :manual:`abortTransaction() page
+         </reference/method/Session.abortTransaction/>`  in the Server manual.
+       |
+
+   * - ``commit_transaction()``
+     - | Commits the active transaction for this session. Returns an
+         error if there is no active transaction for the session or if the
+         transaction was ended. To learn more about
+         this method, see the :manual:`commitTransaction() page
+         </reference/method/Session.commitTransaction/>`  in the Server manual.
+
+   * - ``with_transaction()``
+     - | Starts a transaction on this session and executes a callback once, then
+         commits the transaction. In the event of an exception, this method may retry
+         the commit or the entire transaction, which may invoke the callback multiple 
+         times by a single call to ``with_transaction``.
+       |
+       | **Parameters**: ``callback``, ``read_concern``, ``write_concern``, ``read_preference``, ``max_commit_time_ms``
+       | **Return Type**: ``_T``
+
+   * - ``end_session()``
+     - | Finishes this session. If a transaction has started, this method aborts it.
+         Returns an error if there is no active session to end.
+
+A ``ClientSession`` also has methods to retrieve session
+properties and modify mutable session properties. View the :ref:`API
+documentation <api-docs-transaction>` to learn more about these methods.
```

Comment:
S: Since the style guide recommends that "Link text should begin with the purpose of the cross-reference":

```suggestion
properties and modify mutable session properties. To learn more about these methods, see the :ref:`API
documentation <api-docs-transaction>`.
```